File: Source Code Ver 2.0/quizbackend/src/main/java/vn/hcmuaf/edu/fit/quizbackend/service/QuestionService.java
package vn.hcmuaf.edu.fit.quizbackend.service;

import java.util.Set;

import vn.hcmuaf.edu.fit.quizbackend.model.exam.Question;
import vn.hcmuaf.edu.fit.quizbackend.model.exam.Quiz;

public interface QuestionService {

	public Question addQuestion(Question question);

	public Question updateQuestion(Question question);

	public Set<Question> getQuestions();

	public Question getQuestion(Long questionId);

	public Set<Question> getQuestionsOfQuiz(Quiz quiz);
	
	public void deleteQuestion(Long quesId);
	
	public Question get(Long id);

}

File: Source Code FE 2.0/quizexample/src/app/services/question.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import baseUrl from "./helper";

@Injectable({
  providedIn: 'root'
})
export class QuestionService {

  constructor(private http:HttpClient) { }

  public getQuestionOfQuiz(qid:any){
    return this.http.get(`${baseUrl}/question/quiz/all/${qid}`)
  }

  public getQuestionOfQuizForTest(qid:any){
    return this.http.get(`${baseUrl}/question/quiz/${qid}`)
  }

  public addQuestion(question:any){
    return this.http.post(`${baseUrl}/question/`,question)
  }

  public deleteQuestion(questionId:any){
    return this.http.delete(`${baseUrl}/question/${questionId}`)
  }

  public getQuestion(questionId:any){
    return this.http.get(`${baseUrl}/question/${questionId}`)
  }

  public updateQuestion(question:any){
    return this.http.put(`${baseUrl}/question/`,question)
  }

  public evalQuiz(question:any){
    return this.http.post(`${baseUrl}/question/eval-quiz`,question);
  }

}
